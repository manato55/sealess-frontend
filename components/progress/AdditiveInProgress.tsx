import React, {useEffect, useState} from 'react'
import axios from '../../axios'
import { saveAs } from 'file-saver';
import {useProgress} from '../../hooks/useProgress'

type TopProps = {
    filename: string;
    taskId: number;
}

export const AdditiveInProgress = (props: TopProps) => {
    const [files, setFiles] = useState<string[]>([])
    const {downloadFile} = useProgress()
    
    useEffect(() => {
        const fetchFiles = async() => {
            if(props.filename !== null) {
                const fileTmp = props.filename.split(',')
                setFiles(fileTmp)
            } else {
                setFiles(null)
            }
        }
        fetchFiles()
    },[props.filename]);

    const download = (e: React.MouseEvent<HTMLSpanElement,MouseEvent> ):void => {
        const data:{filename: string; id: number} = {
            filename:e.currentTarget.innerHTML,
            id: props.taskId
        } 
        downloadFile(data)
    }

    return (
        <>
            {files !== null && files.length !== 0 && files.map((file,index) => 
                <ul key={index}>
                    <li onClick={(e) => download(e)}>{file}</li>
                </ul>
            )}
        </>
    )
}


export default AdditiveInProgress