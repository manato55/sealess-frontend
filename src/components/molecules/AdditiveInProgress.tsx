import React, { useEffect, useState } from 'react';
import { useProgress } from '../../hooks/useProgress';
import { saveAs } from 'file-saver';
import { useSetRecoilState } from 'recoil';
import { http } from '../../store/atom';
import styled from 'styled-components';

type TopProps = {
  filename: string;
  taskId: number;
};

export const AdditiveInProgress = (props: TopProps): React.ReactElement => {
  const [files, setFiles] = useState<string[]>([]);
  const { downloadFile } = useProgress();
  const setHttpStatus = useSetRecoilState(http);

  useEffect(() => {
    const fetchFiles = async () => {
      if (props.filename !== null) {
        const fileTmp = props.filename.split(',');
        setFiles(fileTmp);
      } else {
        setFiles(null);
      }
    };
    fetchFiles();
  }, [props.filename]);

  const download = async (e: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
    const data: { filename: string; id: number } = {
      filename: e.currentTarget.innerHTML,
      id: props.taskId,
    };
    const res = await downloadFile(data);
    if (!res.Failure) {
      const blob = new Blob([res.value.data], {
        type: res.value.data.type,
      });
      const contentDisposition = res.value.headers['content-disposition'];
      const fileName = contentDisposition.substring(contentDisposition.indexOf('=') + 1);
      saveAs(blob, fileName);
    } else {
      setHttpStatus(res.error.code);
    }
  };

  return (
    <>
      {files !== null &&
        files.length !== 0 &&
        files.map((file, index) => (
          <ul key={index}>
            <FileTag onClick={(e) => download(e)}>{file}</FileTag>
          </ul>
        ))}
    </>
  );
};

const FileTag = styled.li`
  cursor: pointer;
`;

export default AdditiveInProgress;
