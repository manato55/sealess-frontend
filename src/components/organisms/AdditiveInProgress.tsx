import React, {useEffect, useState} from 'react'
import {useProgress} from '../../hooks/useProgress'

type TopProps = {
    filename: string;
    taskId: number;
}

export const AdditiveInProgress = (props: TopProps): React.ReactElement => {
    const [files, setFiles] = useState<string[]>([])
    const {downloadFile} = useProgress()
    
    useEffect(() => {
        const fetchFiles = async() => {
            if(props.filename !== null) {
                const fileTmp = props.filename.split(',')
                setFiles(fileTmp)
            } else {
                setFiles(null)
            }
        }
        fetchFiles()
    },[props.filename]);

    const download = (e: React.MouseEvent<HTMLSpanElement,MouseEvent> ):void => {
        const data:{filename: string; id: number} = {
            filename:e.currentTarget.innerHTML,
            id: props.taskId
        } 
        downloadFile(data)
    }

    return (
        <>
            {files !== null && files.length !== 0 && files.map((file,index) => 
                <ul key={index}>
                    <li onClick={(e) => download(e)}>{file}</li>
                </ul>
            )}
        </>
    )
}


export default AdditiveInProgress