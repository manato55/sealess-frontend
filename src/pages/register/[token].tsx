import RegisterToken from '../../components/organisms/RegisterToken';

const Token = () => {
  return (
    <>
      <RegisterToken />
    </>
  );
};

export default Token;
