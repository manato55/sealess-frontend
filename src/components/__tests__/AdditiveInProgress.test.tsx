import React from 'react';
import AdditiveInProgress from '../molecules/AdditiveInProgress';
import { RecoilRoot } from 'recoil';
import { mount, shallow } from 'enzyme';
import 'jsdom-global/register';
import { act } from 'react-dom/test-utils';

describe('AdditiveInProgress test', () => {
  let container;

  beforeEach(() => {
    container = mount(
      <RecoilRoot>
        <AdditiveInProgress filename={'testFile'} taskId={3} />
      </RecoilRoot>
    );
  });

  test('click test', () => {
    // const adds = require('../organisms/AdditiveInProgress');
    // jest.spyOn(adds, 'default');
    // container.find('li').simulate('click');
    // adds.default();
    // expect(adds).toHaveBeenCalled();
  });

  test('snapshot test', () => {
    expect(container).toMatchSnapshot();
  });
});
