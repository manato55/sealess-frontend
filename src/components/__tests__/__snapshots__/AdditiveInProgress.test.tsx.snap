// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditiveInProgress test snapshot test 1`] = `
<RecoilRoot>
  <RecoilRoot_INTERNAL>
    <Batcher
      setNotifyBatcherOfChange={[Function]}
    />
    <AdditiveInProgress
      filename="testFile"
      taskId={3}
    >
      <ul
        key="0"
      >
        <li
          onClick={[Function]}
        >
          testFile
        </li>
      </ul>
    </AdditiveInProgress>
  </RecoilRoot_INTERNAL>
</RecoilRoot>
`;
